packages/spacebars-tests/template_tests.js

```diff
@@ -3067,3 +3067,59 @@ Tinytest.add("spacebars-tests - template_tests - inclusion with data remove (#31
   test.isTrue(parentView.isDestroyed);
   test.equal(canonicalizeHtml(div.innerHTML), "<span></span>");
 });
+
+testAsyncMulti("spacebars-tests - template_tests - template-level subscriptions", [
+  function (test, expect) {
+    var tmpl = Template.spacebars_template_test_template_level_subscriptions;
+    var tmplInstance;
+
+    // Make sure the subscriptions stop when the template is destroyed
+    var stopCallback = expect();
+    var stopCallback2 = expect();
+
+    // Make sure the HTML is what we expect when one subscription is ready
+    var checkOneReady = expect(function () {
+      test.equal(canonicalizeHtml(div.innerHTML), "");
+    });
+
+    // Make sure the HTML is what we expect when both subscriptions are ready
+    var checkBothReady = expect(function () {
+      test.equal(canonicalizeHtml(div.innerHTML), "ready! true");
+
+      // This will call the template's destroyed callback
+      Blaze.remove(tmplInstance.view);
+    });
+
+    var subscriptionsFinished = 0;
+
+    // Manually use the subscribe ready callback to make sure the template is
+    // doing the right thing
+    var subscribeCallback = expect(function () {
+      subscriptionsFinished++;
+
+      if (subscriptionsFinished === 1) {
+        Tracker.afterFlush(checkOneReady);
+      }
+
+      if (subscriptionsFinished === 2) {
+        Tracker.afterFlush(checkBothReady);
```

Comment:
(Lost the original comment since you re-wrote the test)

Before we merge this, I'd like to understand why we can't use `Deps.flush`. (Happy to pair on it if that would be helpful)